c_src/bin_nif.c

```diff
@@ -89,6 +90,22 @@ from_list(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
 }
 
 static ERL_NIF_TERM
+length_r(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
```

Comment:
I think lenght_r since it's a simple calculation of bytesize/size might as well just live in erlang land the advantage of having it in C is probably neglectable if not even slower from some overhead (just a guess on the overhead)